packages/caravan-psbt/src/psbtv2/psbtv2.ts

```diff
@@ -787,6 +787,88 @@ export class PsbtV2 extends PsbtV2Maps {
     }
   }
 
+  /**
+   * Sets the sequence number for a specific input in the transaction.
+   *
+   * This private helper method is crucial for implementing RBF and other
+   * sequence-based transaction features. It writes the provided sequence
+   * number as a 32-bit little-endian unsigned integer and stores it in the
+   * appropriate input's map using the PSBT_IN_SEQUENCE key.
+   *
+   * The sequence number has multiple uses in Bitcoin transactions:
+   * 1. Signaling RBF (values < 0xfffffffe)
+   * 2. Enabling nLockTime (values < 0xffffffff)
+   * 3. Relative timelock with BIP68 (if bit 31 is not set)
+   *
+   * According to BIP125 (Opt-in Full Replace-by-Fee Signaling):
+   *
+   * - For a transaction to be considered opt-in RBF, it must have at least
+   *   one input with a sequence number < 0xfffffffe.
+   * - The recommended sequence for RBF is 0xffffffff-2 (0xfffffffd).
+   *
+   * Sequence number meanings:
+   * - = 0xffffffff: Then the transaction is final no matter the nLockTime.
+   * - < 0xfffffffe: Transaction signals for RBF.
+   * - < 0xefffffff : Then the transaction signals BIP68 relative locktime.
+   *
+   * For using nLocktime along with Opt-in RBF, the sequence value
+   * should be between 0xf0000000 and 0xfffffffd.
+   *
+   * Care should be taken when setting sequence numbers to ensure the desired
+   * transaction properties are correctly signaled. Improper use can lead to
+   * unexpected transaction behavior or rejection by the network.
+   *
+   * References:
+   * - BIP125: Opt-in Full Replace-by-Fee Signaling
+   *   https://github.com/bitcoin/bips/blob/master/bip-0125.mediawiki
+   * - BIP68: Relative lock-time using consensus-enforced sequence numbers
+   *   https://github.com/bitcoin/bips/blob/master/bip-0068.mediawiki
+   */
+  public setInputSequence(inputIndex: number, sequence: number) {
+    // Check if the PSBT is ready for the Updater role
+    if (!this.isReadyForUpdater) {
+      throw new Error(
+        "PSBT is not ready for the Updater role. Sequence cannot be changed.",
+      );
+    }
+
+    // Check if the input exists
+    if (inputIndex < 0 || inputIndex >= this.inputMaps.length) {
```

Comment:
Instead of counting the input maps, check against `PSBT_GLOBAL_INPUT_COUNT`. This value is the source of truth for how many inputs exist on the psbt and future work for Transaction Extractor will respect this number.

Reply:
I also think `PSBT_GLOBAL_INPUT_COUNT` and `PSBT_GLOBAL_OUTPUT_COUNT` should never have had a setter, but that's something for another version.

Reply:
Thanks for the review ...

also I had a discussion with Buck where we discussed how functionalities like  `estimateVsize` , `absouteFees` , etc can be added to the psbtv2 class to extend it's functionalities .... 

Also I was working on creating an interactive developer guide for `caravan` ,so I created a tool that uses psbtv2 class to show it's methods , also it can be used to Parse an existing PSBT (so better than  bip174.org , if it supports v2 and global xpubs as Buck pointed out ) ....

You can check it out [here](https://caravan-interactive-developer-guide.vercel.app/psbt) and please share any suggestions that you think can improve it :)

Reply:
chnages [done]([review-changes : PSBTV2](https://github.com/caravan-bitcoin/caravan/pull/114/commits/787be0d872d34c3ea27653d472f428ef344f0b24))